app-release-procedures.rst

```diff
@@ -0,0 +1,298 @@
+    * **Rollback strategy**. Develop plan how to retract release and describe it step-by-step.
+    * **Deployment assistants**. Mention ops persons, so they could consider deployment risks as well.
+
+Use provided `Templates`_ for creating PR description.
+
+[WIP] PR
+--------
+
+Creating "Work In Progress" PR is a common way to get an advice or early
+feedback on your code. Put `[WIP]` at the begging of PR title and
+"**Work In Progress, do not merge**" on top of the description; specify
+original issue, questions and concerns; mention people. If PR involves
+configuration changes (Heroku env, add-ons) then create a rollout plan and
+rollback strategy.
+
+
+Code Review
+===========
+
+PR author is responsible for making sure that PR is getting enough attention.
+If there is no feedback on PR:
+    * for 3-4 hours - ping people in the development group chat,
+    * for 6-7 hours - write PM,
+    * for one day or more - discuss it on the stand-up meeting as a blocker.
+
+When reviewing code consider questions like:
+
+    * Are there any obvious logic errors in the code?
+    * Looking at the requirements, are all cases fully implemented?
+    * Are the new automated tests sufficient for the new code?
+      Do existing automated tests need to be rewritten to account
+      for changes in the code?
+    * Does the new code conform to existing style guidelines?
+
+Try to make sure that all necessary changes are made, including:
+
+    * user interface changes
+    * application code changes
+    * documentation updates
+    * help text revisions
+    * database schema revisions
+    * unit test additions
+    * configuration file updates
+    * build script changes
+
+Code review is a communication process that aims to increase team
+synchronization and expertise as a whole. Don't be afraid to ask a question
+during the review or share ideas. On the other hand, PR author is not obligated
+to accept all ideas or recommendations, by the only exception of
+"change request". Use `# ???:` and `# !!!:` code comment tags to specify places
+in the code for which you need advice.
+
+
+Deployment
+==========
+
+**Deployment window**: between 8 and 17 CEST
+
+Before starting deployment consider if there are no additional pre-deployment
+steps that should be taken. Deployment starts from merging PR into `master`
+branch. Only PR author merges code and does all deployment. In the case of
+need, anyone from development team should be able to deploy by following
+roll-out plan provided in PR description.
+
+PR can be merged into `master` only if it is approved by at least a half
+of the leading project developers and there are no change requests.
+
+Projects leading developers:
+
+    * pyramid_bimt - zupo, dz0ny, karantan
+    * DMON - zupo, dz0ny, karantan
+    * EBN - zupo, dz0ny, karantan
+
+Ops people:
+
+    * dz0ny
+    * gurvankervern
+
+Before deploying make sure that you took care of:
+
+    * Creating backups of backing services (DB storage, ...)
```

Comment:
Yep, the idea is to create a precise list of things that should be backed up. That would allow:
1. to exclude human factor of forgetting to backup something before deploy,
2. to get a review on a rollup and backup plans, so in case if some factors weren't considered they will be pointed out before it's too late,
3.  to reduce the bus factor risk

See [example #2](https://github.com/niteoweb/handbook/commit/1f8f2bfcd5f3c8a23458f80af873dfdab1bfbca1#diff-982fbf119451520f96cf783762fa60fcR241), it has a [list of backing services that should be updated](https://github.com/niteoweb/handbook/commit/1f8f2bfcd5f3c8a23458f80af873dfdab1bfbca1#diff-982fbf119451520f96cf783762fa60fcR266) and each item of that list items should be mentioned in [Backuping steps](https://github.com/niteoweb/handbook/commit/1f8f2bfcd5f3c8a23458f80af873dfdab1bfbca1#diff-982fbf119451520f96cf783762fa60fcR272) and [Rollback strategy](https://github.com/niteoweb/handbook/commit/1f8f2bfcd5f3c8a23458f80af873dfdab1bfbca1#diff-982fbf119451520f96cf783762fa60fcR293)